pkg/render/logstorage.go

```diff
@@ -1419,3 +1395,60 @@ func (es elasticsearchComponent) kibanaPodSecurityPolicy() *policyv1beta1.PodSec
 	psp.GetObjectMeta().SetName("tigera-kibana")
 	return psp
 }
+
+// If user has set ResourceRequirements, replace individual field's default value with user's value.
+// If user provided both Limits and Requests, use them.
+// As ResourceRequirements.Limits must be >= ResourceRequirements.Requests,
+// if user provided just Limits, and Limits is <= default Requests, set Requests value as user's Limits value,
+// if user provided just Requests, and Requests is >= default Limits, set Limits value as user's Requests value.
+func overrideResourceRequirements(defaultReq *corev1.ResourceRequirements, userOverrides corev1.ResourceRequirements) {
+	if _, ok := userOverrides.Limits["cpu"]; ok {
+		defaultReq.Limits["cpu"] = *userOverrides.Limits.Cpu()
+		if _, ok := userOverrides.Requests["cpu"]; !ok && defaultReq.Requests.Cpu().Value() > userOverrides.Limits.Cpu().Value() {
+			defaultReq.Requests["cpu"] = *userOverrides.Limits.Cpu()
```

Comment:
Along the lines of my other "anti override" comments, since you are modifying this default object you lose what was originally there. I see two problems with this:
1. In reviewing this I need to make sure that you are not accidentally comparing the new value of the defaults when you meant to compare the old, none existent values
2. If somebody needs check the original defaults for some other reason as an extension of the abilities here there is a very big chance of a bug being implemented.
3. If there is code outside of this function after this function that for some reason needs the original default values then a bug could be easily introduced.

I think you should create a separate, empty resource and populate the values with either the defaults or the user provided resources.